ultralytics/utils/benchmarks.py

```diff
@@ -84,13 +84,17 @@ def benchmark(
         try:
             # Checks
             if i == 9:  # Edge TPU
-                assert LINUX, "Edge TPU export only supported on Linux"
+                assert not (IS_RASPBERRYPI or IS_JETSON or MACOS or WINDOWS), "Edge TPU export only supported on Linux"
             elif i == 7:  # TF GraphDef
                 assert model.task != "obb", "TensorFlow GraphDef not supported for OBB task"
             elif i in {5, 10}:  # CoreML and TF.js
                 assert MACOS or LINUX, "export only supported on macOS and Linux"
             if i in {3, 5}:  # CoreML and OpenVINO
                 assert not IS_PYTHON_3_12, "CoreML and OpenVINO not supported on Python 3.12"
+            if i in {5}:  # CoreML
```

Comment:
@lakshanthad this is not a correct usage of sets, if you want to determine if i equal 5 then obviously use i == 5

Reply:
my bad here. that is a bad implementation i agree

Reply:
Additionally this is not modifying the existing MACOS check on L90, it's adding extra code that need not exist here

Reply:
@lakshanthad lastly are you sure CoreML export is not supported on RPI and Jetson? This is an export only check, it's not claiming inference is supported. On L90 you can see that Linux supports CoreML export.

Reply:
CoreML export not supported on rpi and jetson. I have tested locally @glenn-jocher

Reply:
@lakshanthad is edge tpu export supported or no?

Reply:
Edge tpu export only supported on X86 Linux.
https://coral.ai/docs/edgetpu/compiler/#system-requirements
![image](https://github.com/ultralytics/ultralytics/assets/20147381/69872654-3a62-4220-b546-a85035cae309)

Reply:
@lakshanthad ok got it, thanks!

I'll try to reduce this to the minimum change then.

Reply:
Thanks @glenn-jocher for the support here :)